src/lib/crypto/kimchi_bindings/js/js_backend/unimplemented.js

```diff
@@ -0,0 +1,38 @@
+/* global _0n, _1n */
+
+// TODO domain_generator
+// this should take an int i <= 32 and return the primitive 2^ith root of unity, i.e. a number w with
+// w^(2^i) = 1, w^(2^(i-1)) = -1
+// computed by taking the 2^32th root and squaring 32-i times
+
+// Provides: caml_pasta_fp_domain_generator
+// Requires: _1n
+function caml_pasta_fp_domain_generator(_i) {
+  return [_1n];
+}
+// Provides: caml_pasta_fq_domain_generator
+// Requires: _1n
+function caml_pasta_fq_domain_generator(_i) {
+  return [_1n];
+}
+
+// TODO verification shifts
+
+// Provides: caml_pasta_fp_plonk_verifier_index_shifts
+// Requires: caml_plonk_verification_shifts_of_rust, _0n
+function caml_pasta_fp_plonk_verifier_index_shifts(_log2_size) {
+  var shifts = {s0: [_0n], s1: [_0n], s2: [_0n], s3: [_0n], s4: [_0n], s5: [_0n], s6: [_0n]};
+  return caml_plonk_verification_shifts_of_rust(shifts);
```

Comment:
Thanks! I ended up eliminating any dependencies of the client sdk on the shifts, and deleting this file. I thought it's cleaner if stuff needed only for snarks aren't accidentally used in parts that don't need them. (I wasn't quite consistent with this.. I did implement the `domain_generator` since I already had everything available to compute it.)

Here's the commit that eliminates dependency on the shifts: https://github.com/MinaProtocol/mina/pull/10654/commits/22694fb406fdcf936cd65ec4308c5a6416413d38